File: backend/app/schemas/ingestion.py
from datetime import datetime

from pydantic import BaseModel


class StartIngestionRequest(BaseModel):
    acquisition_id: str


class StartIngestionResponse(BaseModel):
    ingestion_id: str
    acquisition_id: str
    manifest_id: str
    status: str
    total_documents: int
    stream_url: str


class IngestionRunDetail(BaseModel):
    ingestion_id: str
    acquisition_id: str
    manifest_id: str
    status: str
    started_at: datetime
    completed_at: datetime | None = None
    total_documents: int
    processed: int
    failed: int


class IngestionRunSummary(BaseModel):
    ingestion_id: str
    acquisition_id: str
    manifest_id: str
    status: str
    started_at: datetime
    total_documents: int
    processed: int
    failed: int


class DocumentSummary(BaseModel):
    document_id: str
    source_id: str
    title: str
    status: str
    quality_score: float
    document_type: str
    jurisdiction: str
    regulatory_body: str
    chunk_count: int = 0


class DocumentDetail(BaseModel):
    document_id: str
    source_id: str
    staged_document_id: str
    manifest_id: str
    title: str
    full_text_preview: str  # first 500 chars
    status: str
    quality_score: float
    quality_gates: dict
    curation_notes: list[str]
    effective_date: str | None = None
    jurisdiction: str
    regulatory_body: str
    authority_level: str
    document_type: str
    classification_tags: list[str]
    cross_references: list[str]
    section_count: int
    table_count: int
    chunk_count: int
    created_at: datetime
    curated_at: datetime | None = None


class SectionSummary(BaseModel):
    id: str
    heading: str
    level: int
    text_preview: str
    parent_id: str | None = None


class IndexStats(BaseModel):
    total_chunks: int
    indexed_chunks: int
    total_documents: int
    indexed_documents: int
    by_jurisdiction: dict[str, int]
    by_document_type: dict[str, int]

File: frontend/src/types/ingestion.ts
export interface IngestionRunSummary {
  ingestion_id: string;
  acquisition_id: string;
  manifest_id: string;
  status: string;
  started_at: string;
  total_documents: number;
  processed: number;
  failed: number;
}

export interface IngestionRunDetail {
  ingestion_id: string;
  acquisition_id: string;
  manifest_id: string;
  status: string;
  started_at: string;
  completed_at: string | null;
  total_documents: number;
  processed: number;
  failed: number;
}

export interface StartIngestionResponse {
  ingestion_id: string;
  acquisition_id: string;
  manifest_id: string;
  status: string;
  total_documents: number;
  stream_url: string;
}

export interface DocumentSummary {
  document_id: string;
  source_id: string;
  title: string;
  status: string;
  quality_score: number;
  document_type: string;
  jurisdiction: string;
  regulatory_body: string;
  chunk_count: number;
}

export interface DocumentDetail {
  document_id: string;
  source_id: string;
  staged_document_id: string;
  manifest_id: string;
  title: string;
  full_text_preview: string;
  status: string;
  quality_score: number;
  quality_gates: Record<string, { passed: boolean; detail: string }>;
  curation_notes: string[];
  effective_date: string | null;
  jurisdiction: string;
  regulatory_body: string;
  authority_level: string;
  document_type: string;
  classification_tags: string[];
  cross_references: string[];
  section_count: number;
  table_count: number;
  chunk_count: number;
  created_at: string;
  curated_at: string | null;
}

export interface IndexStats {
  total_chunks: number;
  indexed_chunks: number;
  total_documents: number;
  indexed_documents: number;
  by_jurisdiction: Record<string, number>;
  by_document_type: Record<string, number>;
}

export interface IngestionSSEEvent {
  event: string;
  source_id?: string;
  name?: string;
  status?: string;
  quality_score?: number;
  processed?: number;
  failed?: number;
  total?: number;
  error?: string;
  chunks_indexed?: number;
  run_id?: string;
  total_documents?: number;
  message?: string;
}
